ratis-server/src/main/java/org/apache/ratis/server/impl/ConfigurationManager.java

```diff
@@ -31,17 +34,29 @@
  * entries.
  */
 public class ConfigurationManager {
+  private final RaftPeerId id;
   private final RaftConfigurationImpl initialConf;
   private final NavigableMap<Long, RaftConfigurationImpl> configurations = new TreeMap<>();
   /**
    * The current raft configuration. If configurations is not empty, should be
    * the last entry of the map. Otherwise is initialConf.
    */
   private volatile RaftConfigurationImpl currentConf;
+  /** Cache the peer corresponding to {@link #id}. */
+  private volatile RaftPeer currentPeer;
 
-  ConfigurationManager(RaftConfigurationImpl initialConf) {
+  ConfigurationManager(RaftPeerId id, RaftConfigurationImpl initialConf) {
+    this.id = id;
     this.initialConf = initialConf;
-    this.currentConf = initialConf;
+    setCurrentConf(initialConf);
+  }
+
+  private void setCurrentConf(RaftConfigurationImpl currentConf) {
+    this.currentConf = currentConf;
+    final RaftPeer peer = currentConf.getPeer(id, RaftPeerRole.FOLLOWER, RaftPeerRole.LISTENER);
+    if (peer != null) {
+      this.currentPeer = peer;
+    }
```

Comment:
Why do we need the `!= null` check?  If `currentConf` is replaced with one that returns `peer == null`, will we use the obsolete one as `currentPeer`?

(I'm not familiar with how this works, just thinking about the effects of caching the current peer.)

Reply:
`peer` should be set to the value in the latest conf.   When `id` is not in `currentConf`, do not update `peer`.

Reply:
This is to avoid calling `getRaftConf().getPeer(..)` every time in `RaftServer`, even if the conf is not changed.
```java
//RaftServer
    default RaftPeer getPeer() {
      return Optional.ofNullable(getRaftConf().getPeer(getId(), RaftPeerRole.FOLLOWER, RaftPeerRole.LISTENER))
        .orElseGet(() -> getRaftServer().getPeer());
    }
```

Reply:
Previously if the current conf did not have the peer, we'd fallback to `getRaftServer().getPeer()`.  But with the new code, `getCurrentPeer()` would return the leftover `currentPeer` value.

Sorry if I'm misunderstanding something.

Reply:
`RaftPeer` can be updated (e.g. priority) via setConf.  The change here is better since it will return the most updated peer in that particular group.

The previous code falling back to `getRaftServer().getPeer()` is returning the initial peer for the entire server regardless of the group.

Reply:
Makes sense, thanks for the clarification.

Reply:
It is a good thing that you keep asking questions.  Thanks!